onnxruntime/core/graph/graph.cc

```diff
@@ -4046,6 +4047,229 @@ Status Graph::AddConstantProtoAsInitializer(const ONNX_NAMESPACE::NodeProto& nod
   return Status::OK();
 }
 
+static void RenameSubgraphDependentNames(const InlinedHashMap<std::string, std::string>& name_mapping,
+                                         Graph& graph) {
+  for (auto& node : graph.Nodes()) {
+    if (node.ContainsSubgraph()) {
+      for (auto& [name, subgraph] : node.GetAttributeNameToMutableSubgraphMap()) {
+        RenameSubgraphDependentNames(name_mapping, *subgraph);
+      }
+    }
+
+    // NodeArgs need to be updated
+    auto input_defs = node.MutableInputDefs();
+    for (auto& input_def : node.MutableInputDefs()) {
+      auto hit = name_mapping.find(input_def->Name());
+      if (hit != name_mapping.cend()) {
+        input_def = graph.GetNodeArgIncludingParentGraphs(hit->second);
```

Comment:
regarding the function name, is it reassigning input node args instead of "renaming"?

Reply:
+1. This is a bit unclear to me also.

Reply:
Ok, I guess NodeArgs are unique, and only a pointer is stored, and this is just updating the pointer. So, a minor nit is: why can't the name-mapping actually be a mapping from name to NodeArg-pointer? It looks like in all places, you just need the corresponding NodeArg pointer.

Reply:
The node args for the new name are already created with the correct names, when we get here. So we simply query them and replace. Am I missing something?

Reply:
I am renaming the function. I am worried that there is no specific proposals as to the name.

Reply:
> why can't the name-mapping actually be a mapping from name to NodeArg-pointer? It looks like in all places, you just need the corresponding NodeArg pointer.

or even a `map<NodeArg*, NodeArg*>`?

Reply:
Per offline conversation, Name -> NodeArg* should do it.